packages/x-components/src/components/currency/base-currency.vue

```diff
@@ -83,13 +83,9 @@
      * @internal
      */
     protected get renderedFormat(): string {
-      return (
-        this.format ??
-        this.injectedFormat ??
-        //TODO: add here logger
-        //eslint-disable-next-line no-console
-        console.warn('It is necessary to pass a prop or inject the format')
-      );
+      return 'format' in this.$options.propsData!
+        ? this.format
+        : this.injectedFormat ?? this.format;
```

Comment:
Small suggestion.

If you simply remove the prop default value and make it optional again, and do this change here:
```suggestion
      return this.format ?? this.injectedFormat ?? 'i.iii,dd'
```

Then you should be able to keep using this component easily without the boolean destructure trick.

```html
<BaseCurrency v-bind="{ value: result.price.originalValue, format }" />
```

Reply:
When I thought I was being smart but was being dumb instead. Good catch!